src/main/java/org/checkerframework/specimin/JavaTypeCorrect.java

```diff
@@ -116,11 +116,16 @@ private void updateTypeToChange(String errorMessage) {
 
   /**
    * This method takes the name of a class and converts it to the @ClassGetSimpleName type according
-   * to Checker Framework. If the name is already a simple name, this class will not alter it.
+   * to Checker Framework. If the name is already a simple name, this class will not alter it. To
+   * recall a bit, this method belongs to JavaTypeCorrect, where the main job is to correct the
+   * synthetic return type of unsolved methods. UnsolvedSymbolVisitor has take good note of the
+   * details of those synthetic types, so a simple name should be sufficient,
```

Comment:
This Javadoc comment seems confusing to me, and I'm not sure exactly what it is trying to say. 

As I understand it, the purpose of this method is to take a Java simple or fully-qualified name, and turn it into just the simple name. If that's correct, just that would be sufficient Javadoc. Further, though, the specification should also say the same thing. Right now, the parameter type is just `String`, but I don't think this method works unless the input is [dot-separated](https://checkerframework.org/api/org/checkerframework/checker/signature/qual/DotSeparatedIdentifiers.html).

I assume that adding an annotation to the parameter causes the checker to reject code elsewhere. I'd be okay with leaving the parameter of this method unannotated (if that's easier at call sites) if and only if this class adds some defensive coding to check that the input is actually a dot-separated identifier or a simple name already. As it is, if I passed this an actual arbitrary string (as opposed to a class name), it will not fail but will instead mangle that string, which will be a hard-to-debug problem.

Reply:
Professor, can you give me some suggestions on how to do the defensive check? I did try to do that the other day, but I'm not sure how to really check if a string is dot-separated.

Reply:
Since we have JavaParser available, one option would be to try to parse the string as if it were a Java expression, and then make sure that the resulting CompilationUnit contains a single identifier.

Reply:
Can we instead clarify the Javadoc of this method? Since it takes the type complaining of javac as input (something like "Find /type1/ required /type2/", we can be sure that the input will be a valid string representation of a Java type instead of some ambiguous strings?

Reply:
I think that would be alright. In that case, though, we should add a type annotation indicating that the input is a valid type (and its format); in this case, I think that would be `@FullyQualifiedName` (or whatever). Then, we'll need to suppress a warning when we convert javac's output to that type, but that will be easy for a human to verify.

Reply:
Professor, this method convert a possibly fully-qualified name to a simple name. If the name is simple already, this method will return the name itself. Where should we add the type-qualifiers in this case?

Reply:
I see. In that case, maybe use the `@DotSeparatedIdentifier` annotation (or one of the others that are lubs in the Signature Checker hierarchy)?

Reply:
Got it. Thank you, Professor.